### PR TITLE
Add features to TestHarness already within csvdiff

### LINKED ISSUE #23964: Add existing features within csvdiff to TestHarness
## Reason
Features already within csvdiff can be added to the TestHarness allowing for more flexible testing

## Design
`--ignore-fields` and `--custom-columns` will be added to the parameter set for the TestHarness allowing users to exclude certain columns or only perform csvdiffs on a select subset of columns in csv files.

## Impact
Documentation will show the new features, users shouldn't see any change unless directly entering the new parameters.

### PR DESCRIPTION
**Pull request by vanwdani**
Closes #23964

### CODE REVIEW

File: python/TestHarness/testers/CSVDiff.py [hunk 1]
```diff
@@ -21,6 +21,8 @@ def validParams():
         params.addParam('override_rel_err',   [], "A list of customized relative error tolerances.")
         params.addParam('override_abs_zero',   [], "A list of customized absolute zero tolerances.")
         params.addParam('comparison_file', "Use supplied custom comparison config file.")
+        params.addParam('ignore_columns', [], "A list of variables which will not be included in the CSVDiffs operation.")
```
Comment (GiudGiud) [inline python/TestHarness/testers/CSVDiff.py:24]:
```suggestion
        params.addParam('ignore_columns', [], "A list of variables which will not be included in the comparison.")
```
same for the other one

File: python/TestHarness/testers/CSVDiff.py [hunk 2]
```diff
@@ -71,6 +73,12 @@ def processResultsCommand(self, moose_dir, options):
             if self.specs.isValid('override_abs_zero'):
                 csvdiff.append('--custom-abs-zero %s' % (' '.join(self.specs['override_abs_zero'])))
 
+            if self.specs.isValid('ignore_columns'):
+                csvdiff.append('--ignore-fields %s' % (' '.join(self.specs['ignore_columns'])))
+
+            if self.specs.isValid('custom_columns'):
```
Comment (GiudGiud) [inline python/TestHarness/testers/CSVDiff.py:79]: let s check the arguments passed to make sure they make sense:
an ignored column for example should not be a custom or an override column

Comment (GiudGiud) [reply python/TestHarness/testers/CSVDiff.py:79]: unless CSVDiff already errors on this nicely

**Review by GiudGiud — CHANGES_REQUESTED** [review]
is there a test we can add for this?

**Comment by loganharbour** [comment]
Woah. Hold off on this. This will change with https://github.com/idaholab/moose/pull/23561.

**Review by loganharbour — CHANGES_REQUESTED** [review]
Woah. Hold off on this. This will change with https://github.com/idaholab/moose/pull/23561.

**Comment by vanwdani** [comment]
> Woah. Hold off on this. This will change with #23561.

How long?

File: python/mooseutils/csvdiff.py [hunk 3]
```diff
@@ -289,47 +290,53 @@ def diff(self):
...
-                # if customized tolerances specified use them otherwise
-                # use the default
-                if self.custom_columns:
-                    try:
-                        rel_tol = rel_err_map[key]
-                    except:
-                        rel_tol = self.rel_tol
-                if rel_diff > rel_tol:
-                    self.addError(self.files[1], "The values in column \"%s\" don't match. \n\trelative diff:   %.3e ~ %.3e = %.3e (%.3e)" % (key.strip(),
+                if self.ignore is not None:
```
Comment (GiudGiud) [inline python/mooseutils/csvdiff.py:293]: can you avoid the large diff here?

You could do a
if self.ignore:
  continue

**Comment by GiudGiud** [comment]
@socratesgorilla @loganharbour for an estimate on the other PR timeline please

**Comment by socratesgorilla** [comment]
> @socratesgorilla @loganharbour for an estimate on the other PR timeline please

There's an issue with some stochastic tools tests that someone else is supposed to look at but besides that its essentially ready although I'm not the reviewer

**Comment by GiudGiud** [comment]
ok we can't really merge anything right now until moose/next is fixed anyway. 

If we can't get https://github.com/idaholab/moose/pull/23561 merged in a reasonable amount of time, I think this should be merged and https://github.com/idaholab/moose/pull/23561 will need to adapt to make sure the feature used here is working as desired in SchemaDiff

**Comment by socratesgorilla** [comment]
> ok we can't really merge anything right now until moose/next is fixed anyway.
> 
> If we can't get #23561 merged in a reasonable amount of time, I think this should be merged and #23561 will need to adapt to make sure the feature used here is working as desired in SchemaDiff

Asked zach and he's looking into it today supposedly should be mergeable soon hopefully

**Comment by vanwdani** [comment]
@milljm can you look at this and merge it?

**Review by GiudGiud — CHANGES_REQUESTED** [review]
I m not opposed to merging this (once that last thing has been looked at).
I think the reasonable window for the other PR has passed.

**Review by milljm — DISMISSED** [review]
💯  Thank you for doing this!!

File: python/mooseutils/csvdiff.py [hunk 4]
```diff
@@ -289,6 +290,10 @@ def diff(self):
             for val1, val2 in zip( table1[key], table2[key] ):
                 # if customized tolerances specified use them otherwise
                 # use the default
+                if self.ignore and key in self.ignore:
+                    break
+                else:
+                    continue
```
[the remark this answers is not in the document]
Comment (milljm) [reply python/mooseutils/csvdiff.py:296]: Perhaps, though this recursive for loop has additional `continue`'s and `break`'s later down. Must have been working all this time?

[the remark this answers is not in the document]
Comment (GiudGiud) [reply python/mooseutils/csvdiff.py:296]: the ones below are fine.
We continue when things match to save a little time, we break when things dont match, to error early
```suggestion
                if self.ignore and key in self.ignore:
                    break
```

[the remark this answers is not in the document]
Comment (vanwdani) [reply python/mooseutils/csvdiff.py:296]: Yeah, I agree, the else statement is unnecessary here.  I've removed it.

**Review by GiudGiud — APPROVED** [review]
@loganharbour your review has been addressed as far as I can tell

**Comment by loganharbour** [comment]
All of these changes come with the SchemaDiff change with quite literally a few line change

**Review by loganharbour — APPROVED** [review]

**Comment by GiudGiud** [comment]
Test failures are unrelated afaik

**Comment by GiudGiud** [comment]
Thanks for the contribution!

**Pull request merged**